### ISSUE TITLE
Question about using the pretrained model

### ISSUE THREAD

**Issue by Jantory** [Ariel97-star/FFNet-CoronaryArtery-Segmentation#1]
您好，我想要使用你预先训练好的模型来完成一个作业，但是我在导入模型时遇到了一些问题，好想模型中的一些超参有所修改。不知是否可以从您这里得到一些帮助。万分感谢！
Problem solved, it seems the code for FFNet5 is not compatible with the pre-trained model. The following should be the right one:

class FFNet5(nn.Module):
    """
    Optional based on data set complexity
    The model depth = 5
    """
    def __init__(self, img_ch=3, output_ch=1):
        super(FFNet5, self).__init__()

        self.Maxpool = nn.MaxPool2d(kernel_size=2, stride=2)

        self.Conv1 = single_conv(ch_in=img_ch, ch_out=64)
        self.Conv2 = resconv_block(ch_in=64, ch_out=128)
        self.Conv3 = resconv_block(ch_in=128, ch_out=256)
        self.Conv4 = resconv_block(ch_in=256, ch_out=512)
        self.Conv5 = resconv_block(ch_in=512, ch_out=1024)

        self.center = ASPP(1024, 256)

        self.Up5 = up_conv(ch_in=1024, ch_out=512)
        self.Up_conv5 = conv_block(ch_in=1024, ch_out=512)

        self.Up4 = up_conv(ch_in=512, ch_out=256)
        self.Up_conv4 = conv_block(ch_in=512, ch_out=256)

        self.Up3 = up_conv(ch_in=256, ch_out=128)
        self.Up_conv3 = conv_block(ch_in=256, ch_out=128)

        self.Up2 = up_conv(ch_in=128, ch_out=64)
        self.Up_conv2 = conv_block(ch_in=128, ch_out=64)

        self.Conv1_1x1 = nn.Conv2d(64, output_ch, kernel_size=1, stride=1, padding=0)
        self.Conv2_1x1 = nn.Conv2d(128, output_ch, kernel_size=1, stride=1, padding=0)
        self.Conv3_1x1 = nn.Conv2d(256, output_ch, kernel_size=1, stride=1, padding=0)
        self.Conv4_1x1 = nn.Conv2d(512, output_ch, kernel_size=1, stride=1, padding=0)
        self.Conv5_1x1 = nn.Conv2d(4 * output_ch, output_ch, kernel_size=1, stride=1, padding=0)

    def forward(self, x):
        # encoding path
        x1 = self.Conv1(x)

        x2 = self.Maxpool(x1)
        x2 = self.Conv2(x2)

        x3 = self.Maxpool(x2)
        x3 = self.Conv3(x3)

        x4 = self.Maxpool(x3)
        x4 = self.Conv4(x4)

        x5 = self.Maxpool(x4)
        x5 = self.Conv5(x5)

        x5 = self.center(x5)

        d5 = self.Up5(x5)
        d5 = torch.cat((x4, d5), dim=1)
        d5 = self.Up_conv5(d5)

        d4 = self.Up4(x4)
        d4 = torch.cat((x3, d4), dim=1)
        d4 = self.Up_conv4(d4)

        d3 = self.Up3(d4)
        d3 = torch.cat((x2, d3), dim=1)
        d3 = self.Up_conv3(d3)

        d2 = self.Up2(d3)
        d2 = torch.cat((x1, d2), dim=1)
        d2 = self.Up_conv2(d2)

        p1 = self.Conv1_1x1(d2)
        p2 = F.interpolate(self.Conv2_1x1(d3), size=p1.shape[2:], mode='bilinear', align_corners=False)
        p3 = F.interpolate(self.Conv3_1x1(d4), size=p1.shape[2:], mode='bilinear', align_corners=False)
        p4 = F.interpolate(self.Conv4_1x1(d5), size=p1.shape[2:], mode='bilinear', align_corners=False)

        p = torch.cat((p1, p2, p3, p4), 1)
        p = self.Conv5_1x1(p)

        return p

**Comment by yuta-tech** [comment]
Have you solved the problem yet?
I solved it by removing "module." from the key in the state_dict.

```
model = FFNet5(img_ch=1, output_ch=1)
state_dict = torch.load('FFU_Net-score0.8833.pkl')
new_state_dict = OrderedDict()
for k, v in state_dict.items():
    new_state_dict[k[7:]] = v
model.load_state_dict(new_state_dict)
```

Sorry if this has already been resolved.

**Comment by MiaoMiaoYoung** [comment]
Maybe you can try this
```
class FFNet(nn.Module):
    def __init__(self, img_ch=3, output_ch=1):
        super(FFNet, self).__init__()

        self.Maxpool = nn.MaxPool2d(kernel_size=2, stride=2)

        self.Conv1 = single_conv(ch_in=img_ch, ch_out=64)
        self.Conv2 = resconv_block(ch_in=64, ch_out=128)
        self.Conv3 = resconv_block(ch_in=128, ch_out=256)
        self.Conv4 = resconv_block(ch_in=256, ch_out=512)
        self.Conv5 = resconv_block(ch_in=512, ch_out=1024)

        self.center = ASPP(1024, 256)

        self.Up5 = up_conv(ch_in=1024, ch_out=512)
        self.Up_conv5 = conv_block(ch_in=1024, ch_out=512)

        self.Up4 = up_conv(ch_in=512, ch_out=256)
        self.Up_conv4 = conv_block(ch_in=512, ch_out=256)

        self.Up3 = up_conv(ch_in=256, ch_out=128)
        self.Up_conv3 = conv_block(ch_in=256, ch_out=128)

        self.Up2 = up_conv(ch_in=128, ch_out=64)
        self.Up_conv2 = conv_block(ch_in=128, ch_out=64)

        self.Conv1_1x1 = nn.Conv2d(64, output_ch, kernel_size=1, stride=1, padding=0)
        self.Conv2_1x1 = nn.Conv2d(128, output_ch, kernel_size=1, stride=1, padding=0)
        self.Conv3_1x1 = nn.Conv2d(256, output_ch, kernel_size=1, stride=1, padding=0)
        self.Conv4_1x1 = nn.Conv2d(512, output_ch, kernel_size=1, stride=1, padding=0)
        self.Conv5_1x1 = nn.Conv2d(4 * output_ch, output_ch, kernel_size=1, stride=1, padding=0)

    def forward(self, x):
        # encoding path
        x1 = self.Conv1(x)

        x2 = self.Maxpool(x1)
        x2 = self.Conv2(x2)

        x3 = self.Maxpool(x2)
        x3 = self.Conv3(x3)

        x4 = self.Maxpool(x3)
        x4 = self.Conv4(x4)

        x5 = self.Maxpool(x4)
        x5 = self.Conv5(x5)

        x5 = self.center(x5)

        # decoding + concat path

        d5 = self.Up5(x5)
        d5 = torch.cat((x4, d5), dim=1)
        d5 = self.Up_conv5(d5)

        d4 = self.Up4(d5)
        d4 = torch.cat((x3, d4), dim=1)
        d4 = self.Up_conv4(d4)

        d3 = self.Up3(d4)
        d3 = torch.cat((x2, d3), dim=1)
        d3 = self.Up_conv3(d3)

        d2 = self.Up2(d3)
        d2 = torch.cat((x1, d2), dim=1)
        d2 = self.Up_conv2(d2)

        p1 = self.Conv1_1x1(d2)
        p2 = F.interpolate(self.Conv2_1x1(d3), size=p1.shape[2:], mode='bilinear', align_corners=False)
        p3 = F.interpolate(self.Conv3_1x1(d4), size=p1.shape[2:], mode='bilinear', align_corners=False)
        p4 = F.interpolate(self.Conv4_1x1(d5), size=p1.shape[2:], mode='bilinear', align_corners=False)

        p = torch.cat((p1, p2, p3, p4), 1)
        p = self.Conv5_1x1(p)

        return p

```
